bukuserver/views.py

```diff
@@ -589,114 +576,36 @@ def create_model(self, form):
-            "#46BFBD",
-            "#FDB45C",
-            "#FEDCBA",
-            "#ABCDEF",
-            "#DDDDDD",
-            "#ABCABC",
-            "#4169E1",
-            "#C71585",
-            "#FF4500",
-            "#FEDCBA",
-            "#46BFBD",
-        ]
-        show_netloc_table = False
-        if unique_netloc_len > len(colors):
-            max_netloc_item = len(colors)
-            netloc_colors = colors
-            show_netloc_table = True
-        else:
-            netloc_colors = colors[:unique_netloc_len]
-            max_netloc_item = unique_netloc_len
-        most_common_netlocs = netloc_counter.most_common(max_netloc_item)
-        most_common_netlocs = [
-            [val[0], val[1], netloc_colors[idx]]
-            for idx, val in enumerate(most_common_netlocs)
-        ]
-
-        unique_tag_len = len(tag_counter)
-        show_tag_rank_table = False
-        if unique_tag_len > len(colors):
-            max_tag_item = len(colors)
-            tag_colors = colors
-            show_tag_rank_table = True
-        else:
-            tag_colors = colors[:unique_tag_len]
-            max_tag_item = unique_tag_len
-        most_common_tags = tag_counter.most_common(max_tag_item)
-        most_common_tags = [
-            [val[0], val[1], tag_colors[idx]]
-            for idx, val in enumerate(most_common_tags)
-        ]
-
-        unique_title_len = len(title_counter)
-        show_title_rank_table = False
-        if unique_title_len > len(colors):
-            max_title_item = len(colors)
-            title_colors = colors
-            show_title_rank_table = True
-        else:
-            title_colors = colors[:unique_title_len]
-            max_title_item = unique_title_len
-        most_common_titles = title_counter.most_common(max_title_item)
-        most_common_titles = [
-            [val[0], val[1], title_colors[idx]]
-            for idx, val in enumerate(most_common_titles)
-        ]
 
+        datetime = data['generated']
         return self.render(
-            "bukuserver/statistic.html",
-            most_common_netlocs=most_common_netlocs,
-            netloc_counter=netloc_counter,
-            show_netloc_table=show_netloc_table,
-            most_common_tags=most_common_tags,
-            tag_counter=tag_counter,
-            show_tag_rank_table=show_tag_rank_table,
-            most_common_titles=most_common_titles,
-            title_counter=title_counter,
-            show_title_rank_table=show_title_rank_table,
-            datetime=statistic_datetime,
-            datetime_text=statistic_datetime.humanize(
-                arrow.now(), granularity="second"
-            ),
+            'bukuserver/statistic.html',
+            netlocs=CountedData(data['netlocs']),
+            tags=CountedData(data['tags']),
+            titles=CountedData(data['titles']),
+            datetime=datetime,
+            datetime_text=datetime.humanize(arrow.now(), granularity='second'),
         )
```

Comment:
Refactoring stats code to simplify it and remove logic duplication.

Also moved amount filtering here from the UI, and modified it slightly (there's no need to exclude netlocs with 1 URL, but the only good reason to have a "ranking" for _bookmark titles_ is looking for duplicates)